packages/libs/eda/src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -442,6 +451,85 @@ function HistogramViz(props: VisualizationProps<Options>) {
   );
 
+  // get distribution data
+  const subsettingClient = useSubsettingClient();
+
+  const getData = useCallback(async () => {
+    if (vizConfig.xAxisVariable != null && xAxisVariable != null) {
+      const [displayRangeMin, displayRangeMax, binWidth, binUnits] =
+        NumberVariable.is(xAxisVariable)
+          ? [
+              xAxisVariable.distributionDefaults.displayRangeMin ??
+                xAxisVariable.distributionDefaults.rangeMin,
+              xAxisVariable.distributionDefaults.displayRangeMax ??
+                xAxisVariable.distributionDefaults.rangeMax,
+              xAxisVariable.distributionDefaults.binWidth,
+              undefined,
+            ]
+          : [
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMin ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMin,
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMax ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMax,
+              (xAxisVariable as DateVariable).distributionDefaults.binWidth,
+              (xAxisVariable as DateVariable).distributionDefaults.binUnits,
+            ];
+
+      const distribution = await getDistribution<DistributionResponse>(
+        {
+          entityId: vizConfig.xAxisVariable?.entityId ?? '',
+          variableId: vizConfig.xAxisVariable?.variableId ?? '',
+          filters: filters,
+        },
+        (filters) => {
+          return subsettingClient.getDistribution(
+            studyMetadata.id,
+            vizConfig.xAxisVariable?.entityId ?? '',
+            vizConfig.xAxisVariable?.variableId ?? '',
+            {
+              valueSpec: 'count',
+              filters,
+              binSpec: {
+                // Note: technically any arbitrary values can be used here for displayRangeMin/Max
+                // but used more accurate value anyway
+                displayRangeMin: DateVariable.is(xAxisVariable)
+                  ? displayRangeMin + 'T00:00:00Z'
+                  : displayRangeMin,
+                displayRangeMax: DateVariable.is(xAxisVariable)
+                  ? displayRangeMax + 'T00:00:00Z'
+                  : displayRangeMax,
+                binWidth: binWidth ?? 1,
+                binUnits: binUnits,
+              },
+            }
+          );
+        }
+      );
+
+      // return series using foreground response
+      const series = {
+        series: [
+          distributionResponseToDataSeries(
+            'Subset',
+            distribution.foreground,
+            red,
+            NumberVariable.is(xAxisVariable) ? 'number' : 'date'
+          ),
+        ],
+      };
+
+      return series;
+    }
+
+    return undefined;
+  }, [filters, xAxisVariable, vizConfig.xAxisVariable]);
+
+  const getDistributionData = usePromise(
+    useCallback(() => getData(), [getData])
```

Comment:
`getData` is already wrapped in a `useCallback` - I don't think it needs another one.